url86119f019a_solzy_hive_parse_tgz-pJ8-org_apache_hadoop_hive_ql_parse_MainJ8/src/org/apache/hadoop/hive/ql/parse/HiveLexer.java

```diff
@@ -0,0 +1,15495 @@
+				if ( (LA71_1=='o') ) {
+					int LA71_2 = input.LA(3);
+					if ( (LA71_2=='u') ) {
+						int LA71_4 = input.LA(4);
+						if ( (LA71_4=='r') ) {
+							int LA71_7 = input.LA(5);
+							if ( (LA71_7=='s') ) {
+								alt71=2;
+							}
+
+							else {
+								alt71=1;
+							}
+
+						}
+
+						else {
+							int nvaeMark = input.mark();
+							try {
+								for (int nvaeConsume = 0; nvaeConsume < 4 - 1; nvaeConsume++) {
+									input.consume();
+								}
+								NoViableAltException nvae =
+									new NoViableAltException("", 71, 4, input);
+								throw nvae;
+							} finally {
+								input.rewind(nvaeMark);
+							}
+						}
+
+					}
+
+					else {
+						int nvaeMark = input.mark();
+						try {
+							for (int nvaeConsume = 0; nvaeConsume < 3 - 1; nvaeConsume++) {
+								input.consume();
+							}
+							NoViableAltException nvae =
+								new NoViableAltException("", 71, 2, input);
+							throw nvae;
+						} finally {
+							input.rewind(nvaeMark);
+						}
+					}
+
+				}
+				else if ( (LA71_1=='r') ) {
+					int LA71_3 = input.LA(3);
+					if ( (LA71_3=='s') ) {
+						alt71=4;
+					}
+
+					else {
+						alt71=3;
+					}
+
+				}
+
+				else {
+					int nvaeMark = input.mark();
+					try {
+						input.consume();
+						NoViableAltException nvae =
+							new NoViableAltException("", 71, 1, input);
+						throw nvae;
+					} finally {
+						input.rewind(nvaeMark);
+					}
+				}
+
+			}
+
+			else {
+				NoViableAltException nvae =
+					new NoViableAltException("", 71, 0, input);
+				throw nvae;
+			}
+
+			switch (alt71) {
+				case 1 :
+					// HiveLexer.g:336:10: 'hour'
+					{
+					match("hour"); 
+
+					}
+					break;
+				case 2 :
+					// HiveLexer.g:336:21: 'hours'
+					{
+					match("hours"); 
+
+					}
+					break;
+				case 3 :
+					// HiveLexer.g:336:33: 'hr' ( DOT )?
+					{
+					match("hr"); 
+
+					// HiveLexer.g:336:38: ( DOT )?
+					int alt69=2;
+					int LA69_0 = input.LA(1);
+					if ( (LA69_0=='.') ) {
+						alt69=1;
+					}
+					switch (alt69) {
+						case 1 :
+							// HiveLexer.g:
+							{
+							if ( input.LA(1)=='.' ) {
+								input.consume();
+							}
+							else {
+								MismatchedSetException mse = new MismatchedSetException(null,input);
+								recover(mse);
+								throw mse;
+							}
+							}
+							break;
+
+					}
+
+					}
+					break;
+				case 4 :
+					// HiveLexer.g:336:46: 'hrs' ( DOT )?
+					{
+					match("hrs"); 
+
+					// HiveLexer.g:336:52: ( DOT )?
+					int alt70=2;
+					int LA70_0 = input.LA(1);
+					if ( (LA70_0=='.') ) {
+						alt70=1;
+					}
+					switch (alt70) {
+						case 1 :
+							// HiveLexer.g:
+							{
+							if ( input.LA(1)=='.' ) {
+								input.consume();
+							}
+							else {
+								MismatchedSetException mse = new MismatchedSetException(null,input);
+								recover(mse);
+								throw mse;
+							}
+							}
+							break;
+
+					}
+
+					}
+					break;
+
+			}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 8966:9042, 9161:9237.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.